### PR TITLE
Remove unsupported cuda versions

### PR DESCRIPTION
**Pull request by sevagh**
This PR removes unsupported CUDA toolkit versions from the gpuci-build-environment matrix.

### CODE REVIEW

File: ci/axis/miniforge-cuda-arm64.yml [hunk 1]
```diff
@@ -20,7 +20,6 @@ ARCH_TYPE:
 CUDA_VER:
   - 11.5.1
   - 11.4.1
-  - 11.3.1
```
Comment (ajschmidt8) [inline ci/axis/miniforge-cuda-arm64.yml:23]: there are some more `11.3.1` references towards the bottom of this file that can be removed.

File: ci/axis/miniforge-cuda-driver-arm64.yml [hunk 2]
```diff
@@ -13,7 +13,6 @@ DOCKER_FILE:
 CUDA_VER:
   - 11.5
   - 11.4
-  - 11.3
```
Comment (ajschmidt8) [inline ci/axis/miniforge-cuda-driver-arm64.yml:16]: CUDA 11.3 is associated with driver 465 (see table 3 [here](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html)). so any mentions of driver 465 in this file should also be removed.

File: ci/axis/miniforge-cuda-driver.yml [hunk 3]
```diff
@@ -14,9 +14,7 @@ DOCKER_FILE:
 CUDA_VER:
   - 11.5
   - 11.4
-  - 11.3
```
Comment (ajschmidt8) [inline ci/axis/miniforge-cuda-driver.yml:17]: CUDA 11.3 is associated with driver 465 (see table 3 [here](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html)). so any mentions of driver 465 in this file should also be removed.

File: ci/axis/miniforge-cuda-driver.yml [hunk 4]
```diff
@@ -14,9 +14,7 @@ DOCKER_FILE:
...
   - 11.2
-  - 11.1
```
Comment (ajschmidt8) [inline ci/axis/miniforge-cuda-driver.yml:19]: CUDA 11.1 is associated with driver 455 (see table 3 [here](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html)). so any mentions of driver 455 in this file should also be removed.

File: ci/axis/rapidsai-driver.yml [hunk 5]
```diff
@@ -16,9 +16,7 @@ RAPIDS_VER:
 CUDA_VER:
   - 11.5
   - 11.4
-  - 11.3
```
Comment (ajschmidt8) [inline ci/axis/rapidsai-driver.yml:19]: same comments about the driver versions for this file. let's remove all 455 and 465 mentions.

**Review by raydouglass — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**